### PR TITLE
[couch-to-sql] [repeaters] add connection settings in SqlRepeater

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Summary

https://dimagi-dev.atlassian.net/browse/SAAS-12155
Since both Repeaters and ConnectionSettings would be in SQL so it would make sense to have a ForeignKey instead of id.
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/motech/repeaters/models.py [hunk 1]
```diff
@@ -196,17 +196,14 @@ def all_ready(self):
                 .filter(repeat_records_ready_to_send))
 
 
-class RepeaterStub(models.Model):
-    """
-    This model links the SQLRepeatRecords of a Repeater.
-    """
+class SQLRepeater(models.Model):
```
Comment (millerdev) [inline corehq/motech/repeaters/models.py:199]: The typical pattern is to specify a name without the `sql_` prefix in `Meta.db_table`. Would that be possible here?

See the `db_table` notes in https://commcare-hq.readthedocs.io/couch_to_sql_models.html#pr-1-add-sql-model-and-migration-management-command-write-to-sql

File: corehq/motech/repeaters/models.py [hunk 2]
```diff
@@ -196,23 +196,26 @@ def all_ready(self):
...
     is_paused = models.BooleanField(default=False)
     next_attempt_at = models.DateTimeField(null=True, blank=True)
     last_attempt_at = models.DateTimeField(null=True, blank=True)
 
-    objects = RepeaterStubManager()
+    objects = RepeaterManager()
+
+    connection_settings = models.ForeignKey(
+        ConnectionSettings,
+        on_delete=models.PROTECT
```
Comment (AmitPhulera) [inline corehq/motech/repeaters/models.py:210]: In my opinion, `PROTECT` strategy should be optimum here. But are there any use cases where we want to allow that ConnectionSettings instance referring to a Repeater to be deleted?

**Review by millerdev — APPROVED** [review]
It's probably worth squashing the three migrations in this PR into a single migration (I think the easiest way to do this is to delete them all and re-run `makemigrations`). More migrations makes test setup and other things take longer, so fewer is better.

File: corehq/motech/repeaters/models.py (hunk 2 above)
Comment (kaapstorm) [reply corehq/motech/repeaters/models.py:210]: You are right. We don't want ConnectionSettings for a Repeater to be deleted before the Repeater is deleted.

**Pull request merged**